### ISSUE TITLE
Unresolved External Symbol Detours--- referenced in function

### ISSUE THREAD

**Issue by ZhongDev** [SirusDoma/MaiSense#4]
I cant seem to build MaiSense properly.
I've put all the `.h` files in `extlibs\include` into `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include`
and the 
`detours.lib` file in the `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\lib\x86`
(did this because when i tried to link via the project preferences it wouldn't find detours.h at all; "`cannot open source file Detours.h`")

![image](https://user-images.githubusercontent.com/47954501/75928596-dcd78680-5ec2-11ea-9a93-72aab89c33ad.png)

If I could get some guidance on this that would be great.
Thanks.

**Comment by andylizi** [comment]
`MaiSense.Input` Properties -> `Linker` -> `Input` -> Set `Additional Dependencies` to `detours.lib`
![](https://user-images.githubusercontent.com/12008103/75933725-a1828b00-5eb5-11ea-966c-b69fc0d5145e.png)

If you see `LNK1106 cannot open file 'detours.lib'` error, set `Additional Library Properties` in `Linker` -> `General` to `./extlibs/lib/;%(AdditionalLibraryDirectories)`.

**Comment by ZhongDev** [comment]
The libraries were already linked before. (still getting same errors as initial post after setting the fields outlined)
I did change the Platform from x64 to Win32, and it successfully built the project, but it wouldn't inject properly; i.e program crashes after the initial loading screen after `Score: Good`

**Comment by andylizi** [comment]
> program crashes after the initial loading screen after `Score: Good`

I had tried to solve exactly the same problem for the longest time, and after hours and hours of debugging I realized that, for some bizarre reason, the code on the Github is not the same as in the released binary version - the latter works, the former does not WHATSOEVER. More specifically, the built-in address of Maimai's `GameInput` function is different. So when the program is trying to hook the game function using detours, it's operating on the wrong part of the memory, and the game will crash when it finished loading.

I had to use IDA to reverse-engineer the released binary and extract the correct address `0x00571610`. Update the following line to reflect this and it will work.

https://github.com/SirusDoma/MaiSense.API/blob/e5c4295caa929fd5154adbef69ad4bfa641f975b/src/Input/InputManager.cpp#L27
